cmd/zed/agents/zfs_retire.c

```diff
@@ -213,16 +270,60 @@ replace_with_spare(fmd_hdl_t *hdl, zpool_handle_t *zhp, nvlist_t *vdev)
 		fmd_hdl_debug(hdl, "zpool_vdev_replace '%s' with spare '%s'",
 		    dev_name, basename(spare_name));
 
-		if (zpool_vdev_attach(zhp, dev_name, spare_name,
-		    replacement, B_TRUE) == 0)
+		ret = zpool_vdev_attach(zhp, dev_name, spare_name,
+		    replacement, B_TRUE);
+		if (ret == 0)
 			break;
+
+		if (libzfs_errno(zpool_get_handle(zhp)) == EZFS_BUSY) {
+			fmd_hdl_debug(hdl,
+			    "zpool_vdev_attach '%s' busy. Saving request.'",
+			    dev_name);
```

Comment:
nit: I'd suggest removing this debug line since there is a similar one in `save_deferred_spare()` which must succeed, it can provide more information, and this is the only caller of `save_deferred_spare()` so there no ambiguity about how you got there..  You might want to extend it to include the `dev_name`.